include/integratorxx/quadratures/radial/lmg.hpp

```diff
@@ -0,0 +1,127 @@
+#include <integratorxx/util/lambert_w.hpp>
+#include <integratorxx/quadratures/radial/radial_transform.hpp>
+
+namespace IntegratorXX {
+namespace lmg {
+
+// Eq 19 of LMG paper (DOI 10.1007/s002140100263)
+inline double r_upper_obj(int m, double alpha, double r) {
+  const double am_term = (m + 1.0) / 2.0;
+  const double g_term  = std::tgamma((m + 3.0) / 2.0);
+  const double x = alpha * r * r;
+  return g_term * std::pow(x, am_term) * std::exp(-x);
+}
+
+// Solve Eq 19 of LMG paper (DOI 10.1007/s002140100263)
+inline double r_upper(int m, double alpha, double prec) {
+  // P = G * (ALPHA * R^2)^((M+1)/2) * EXP(-ALPHA * R^2)
+  // P = G  * X^L * EXP(-X):   X = ALPHA * R^2, L = (M+1)/2
+  // X = -L * LAMBERT_W( - (P/G)^(1/L) / L )
+  // R = SQRT(X / ALPHA)
```

Comment:
Is this the OpenMolcas source?

Reply:
No, this is just my internal notes to remember how I got to the expression - OpenMolcas (and `numgrid` for that matter) use a stepping procedure to get to this result. OpenMolcas' is very approximate, while `numgrid`'s is much closer to FP64 precision. My newton implementation of `lambert_w` (based on the [Knuth algorithm](https://doi.org/10.1007%2FBF02124750)) converges much quicker than `numgrid` and has a better guaranteed accuracy